README.md

```diff
@@ -0,0 +1,53 @@
+# Pre-initialize WebAssembly Components
+
+[Wizer][] demonstrated that WebAssembly modules can be made to start up
+faster by pre-initializing them. `component-init` does the same thing,
+but for the WebAssembly Component Model.
+
+[Wizer]: https://github.com/bytecodealliance/wizer/
+
+`component-init` is a Rust library that can be used in other projects.
+(One prominent example is [componentize-py][].) To use it in your
+project, you additionally need a WebAssembly runtime that supports the
+Component Model; [Wasmtime][] is known to work.
+
+[componentize-py]: https://github.com/bytecodealliance/componentize-py
+[Wasmtime]: https://wasmtime.dev/
+
+# Design rationale
+
+To pre-initialize a WebAssembly program, you need to do three things:
+
+1. Run its initialization function.
+2. Snapshot any resulting changes to its internal mutable state.
+3. Emit a new WebAssembly program with the updated mutable state.
+
+Step 1 is easy and any runtime will do. Step 3 is a little tedious but
+it just involves copying the original program mostly unchanged except
+for updating data sections and such, so it's not too hard.
+
+The challenge is in step 2: How do you refer to the internal state from
+outside the program? Some state is explicitly exported from a module or
+component, so that's easy to access via any runtime, but what about the
+rest?
+
+To solve this, both Wizer and component-init insert an extra step at the
```

Comment:
Does Wizer actually do this?  I had assumed it used special host powers to look at the runtime state of the instantiated module without modifying module it was instantiated from first, but I haven't looked to see if that's true.

Reply:
I based my description on a quick look at https://github.com/bytecodealliance/wizer/blob/main/src/instrument.rs and especially the doc comment near the top of the file. It looks to me like Wizer just exports globals and memories directly under names of its choice, that it can use normal public Wasmtime APIs to read later.